src/instructions.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include "instructions.hpp"
+uint16_t Instructions::combine8BitTo16Bit(uint8_t lhs, uint8_t rhs)
+{
+    return ((uint16_t)lhs << 8) | rhs;
+}
+
+void Instructions::load16NextBitToRegister(IMemory::REG16BIT reg, IMemory& memory)
+{
+    uint16_t cursor = memory.get16BitRegister(IMemory::REG16BIT::PC);
+    IMemory::RomData const & rom = memory.getReadOnlyMemory();
+    uint16_t value = combine8BitTo16Bit(rom[cursor + 2],rom[cursor + 1]);
+    memory.set16BitRegister(reg, value);
+    memory.set16BitRegister(IMemory::REG16BIT::PC, cursor + 3);
+}
+
+void Instructions::load8BitInRegisterAtAdress(IMemory::REG8BIT reg, IMemory::REG16BIT adress, IMemory& memory)
+{
+    uint8_t valueRegister = memory.get8BitRegister(reg);
+    uint16_t valueAdress = memory.get16BitRegister(adress);
+    memory.writeInROM(valueRegister, valueAdress);
+    uint16_t cursor = memory.get16BitRegister(IMemory::REG16BIT::PC);
+    memory.set16BitRegister(IMemory::REG16BIT::PC, ++cursor);
+}
+
+void Instructions::xor8BitRegister(IMemory::REG8BIT reg, IMemory& memory)
+{
+    uint16_t cursor = memory.get16BitRegister(IMemory::REG16BIT::PC);
+    memory.set8BitRegister(reg, 0);
+    memory.set16BitRegister(IMemory::REG16BIT::PC, ++cursor);
+}
+
+int Instructions::load16BitToHL(IMemory& memory)
+{
+    load16NextBitToRegister(IMemory::REG16BIT::HL, memory);
+    return 8; //Cycles
```

Comment:
I think those cycles should be part of your map _instructions.
instead of having this :   
```  
std::map<uint8_t, funcInstruct> _instructions =
  {            {0x21, std::bind(&Instructions::load16BitToHL, this, _1)}   }
```
you could do this : 
```
struct instructionData {
   int cycle = 0;
   funcInstruct  handler;
} instructionData_t;

std::map<uint8_t, instructionData _t> _instructions =
  {            {0x21, { 4 /* number of cycle */  , std::bind(&Instructions::load16BitToHL, this, _1)} }  }
```

Putting the cycle in the map will be easier for maintenance, to populate, to make changes and easier to compare to spot issues. :)